Sources/ApexyAlamofire/AlamofireClient+Concurrency.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  AlamofireClient+Concurrency.swift
+//  
+//
+//  Created by Aleksei Tiurnin on 15.08.2022.
+//
+
+import Alamofire
+import Apexy
+import Foundation
+
+@available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+extension AlamofireClient: ConcurrencyClient {
+    
+    func observeResponse(
+        dataResponse: DataResponse<Data, AFError>,
+        error: Error?) async {
+            return await withCheckedContinuation{ continuation in
+                completionQueue.async {[weak self] in
+                    self?.responseObserver?(
+                        dataResponse.request,
+                        dataResponse.response,
+                        dataResponse.data,
+                        error)
+                    continuation.resume()
+                }
+            }
+        }
+    
+    open func request<T>(_ endpoint: T) async throws -> T.Content where T : Endpoint {
+        
+        let anyRequest = AnyRequest(create: endpoint.makeRequest)
+        let request = sessionManager.request(anyRequest)
+            .validate { request, response, data in
+                Result(catching: { try endpoint.validate(request, response: response, data: data) })
+            }
+
+        let dataResponse = await request.serializingData().response
+        let result = APIResult<T.Content>(catching: { () throws -> T.Content in
+            do {
+                let data = try dataResponse.result.get()
+                return try endpoint.content(from: dataResponse.response, with: data)
+            } catch {
+                throw error.unwrapAlamofireValidationError()
+            }
+        })
+
+        Task.detached {[weak self, dataResponse, result] in
```

Comment:
I didn't understand why to do a `Task.detach` that calls a method inside which `completionQueue.async` occurs.

Thus, there are two switches to other threads